builtin/mainmenu/settings/shader_warning_component.lua

```diff
@@ -0,0 +1,26 @@
+-- Minetest
+-- SPDX-License-Identifier: LGPL-2.1-or-later
+
+return {
+	query_text = "Shaders",
+	requires = {
+		shaders_support = true,
+        shaders = false,
+	},
+    full_width = true,
+	get_formspec = function(self, avail_w)
+        local fs = {
+            "box[0,0;", avail_w, ",1.2;", mt_color_orange, "]",
+			"label[0.2,0.4;", fgettext("Shaders are disabled."), "]",
+			"label[0.2,0.8;", fgettext("This is not a recommended configuration."), "]",
+			"button[", avail_w - 2.2, ",0.2;2,0.8;fix_shader_warning;Enable]",
+        }
+		return table.concat(fs, ""), 1.2
+	end,
+	on_submit = function(self, fields)
+		if fields.fix_shader_warning then
+			core.settings:remove("enable_shaders") -- default value is true
```

Comment:
Wouldn't it be better to directly set this to true in order to avoid surprises in case the default value changes at some point?

Reply:
1. The default value of `enable_shaders` won't change to false. If it did, showing a warning when `enable_shaders` is false would make no sense anyway.
2. It's better not to clutter the configuration with useless settings

Reply:
fair enough.